### PR TITLE
Release 1.3.2 branch commit

### PR DESCRIPTION
**Pull request by smartguest**
Adds in the feature for change password and also changes made by Charles and Cheena. 

I am not sure how to handle the fact that azdata currently does not have the PasswordChangeResult interface in node modules (leading to a "cannot be found" error in vscode), but the dataprotocol itself works when copied directly to the MSSQL extension. If I should address this is any way, please let me know how to do so.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
 export interface CreateSessionResponse {
+	/**
+	 * Unique Id to use when sending any requests for objects in the tree
+	 * under the node
+	 */
 	sessionId: string;
 }
 
+/**
+ * Information returned from a createSessionRequest. Contains success information, a sessionId to be used
```
Comment (kburtram) [inline src/types.ts:16]: It's seems incorrect to return a parameters object from a function.  Typically, parameters are input to the method and a response/result object is the output.  As such, the comment seems incorrect (or the interface name) and the error fields seem out of place.

File: src/types.ts [hunk 2]
```diff
@@ -1,14 +1,42 @@
 import * as azdata from 'azdata';
 
+/**
+ * Contains success information, a sessionId to be used when requesting
```
Comment (kburtram) [inline src/types.ts:4]: comment mentions values that don't appear to be included in the interface definition (e.g. success info & root node).

File: package.json [hunk 3]
```diff
@@ -15,8 +15,9 @@
     "vscode-languageclient": "5.2.1"
   },
   "devDependencies": {
+    "@types/azdata": "^1.40.0",
```
Comment (kburtram) [inline package.json:18]: Do you know why we're adding this NPM types module and including a copy of the azdata .d.ts file directly in the project.  Typically, you'd do one or other depending on if you need API changes that aren't included in the published module.

File: src/main.ts [hunk 4]
```diff
@@ -295,7 +313,8 @@ export class ConnectionFeature extends SqlOpsFeature<undefined> {
 			rebuildIntelliSenseCache,
 			registerOnConnectionChanged,
 			registerOnIntelliSenseCacheComplete,
-			registerOnConnectionComplete
+			registerOnConnectionComplete,
+			changePassword
```
Comment (kburtram) [inline src/main.ts:317]: nit: it looks like the methods are arranged with "normal" methods first and registration methods at the end.  Probably when updating it makes sense to following the existing convention in the code, e.g., adding this method on line 310 after the other "change" method.

**Review by kburtram — CHANGES_REQUESTED** [review]
There are some minor changes I mentioned in the PR that should be made prior to merging.  Overall, looks fine.  It's too bad that since this is currently MSSQL specific that the changes need to be made in this component (rather than in the MSSQL extension contracts) but since this is attached to the connection provider this is the supported way to do this.  Maybe in the future we should have a way for extensions to add provider-specific APIs on to the common DMP interfaces.

File: package.json (hunk 3 above)
Comment (kburtram) [reply package.json:18]: On second look this looks fine.  It looks like you're removing the checked in azdata.d.ts and consuming from npm, but updating the proposed.d.ts file to get new interfaces.  This is the correct pattern that we should use here.

File: src/types.ts (hunk 1 above)
Comment (smartguest) [reply src/types.ts:16]: These aren't my changes, they are from Cheena's PR.

File: src/types.ts (hunk 2 above)
Comment (smartguest) [reply src/types.ts:4]: See here: https://github.com/microsoft/sqlops-dataprotocolclient/pull/75#discussion_r1046273032

File: package.json (hunk 3 above)
Comment (smartguest) [reply package.json:18]: This is from Charles' PR, not mine.

**Comment by smartguest** [comment]
@cheenamalhotra Can you explain your changes to DataProtocolClient here? Karl has some questions about them. Thanks.

**Review by kburtram — APPROVED** [review]

**Pull request merged**